rhel7/fixes/bash/grub2_enable_fips_mode.sh

```diff
@@ -16,19 +16,23 @@ package_command install dracut-fips
 
 dracut -f
 
-if [ -e /sys/firmware/efi ]; then
-	BOOT=`df /boot/efi | tail -1 | awk '{print $1 }'`
-else
-	BOOT=`df /boot | tail -1 | awk '{ print $1 }'`
-fi
-
 # Correct the form of default kernel command line in /etc/default/grub
 if ! grep -q "^GRUB_CMDLINE_LINUX=\".*fips=1.*\"" /etc/default/grub;
 then
   # Append 'fips=1' argument to /etc/default/grub (if not present yet)
   sed -i "s/\(GRUB_CMDLINE_LINUX=\)\"\(.*\)\"/\1\"\2 fips=1\"/" /etc/default/grub
 fi
 
-# Edit runtime setting
+# Get the UUID of the device mounted at /boot.
+BOOT_DEV=$(df /boot/ | tail -1 | awk '{print $1}')
+BOOT_DEV=$(lsblk -no UUID ${BOOT_DEV})
+
+# Set boot= argument appropriately so dracut's FIPS module can find vmlinuz HMAC files.
+if grep -q "^GRUB_CMDLINE_LINUX=\".*boot=.*\"" /etc/default/grub; then
+        sed -i "s/\(GRUB_CMDLINE_LINUX=\)\"\(.*\)boot=[^s]*\(.*\)\"/\1\"\2 boot=UUID=${BOOT_DEV} \3\"/" /etc/default/grub
```

Comment:
Getting rid of double quotes here may be tricky as there are variables to be expanded, so it's totally up to you whether to attempt to solve it or not.

Reply:
Yeah anytime you're expanding a shell variable in a regex that matches on quotes is going to look bad. 

Also this review made me realize there is an error "[^s]" is completely wrong. Replaced that with POSIX :space: syntax and it probably should match ^GRUB instead of GRUB. Also switched to pipes instead of / for the sed switch since the back slashes for string capture made it look cluttered.

```
if grep -q '^GRUB_CMDLINE_LINUX=".*boot=.*"' /etc/default/grub; then
        sed -i 's|\(^GRUB_CMDLINE_LINUX=.*\)boot=[^[:space:]]*\(.*\)"|\1 boot=UUID='"${BOOT_DEV}"' \2|' /etc/default/grub
```

I'll update the PR with this change if you think it is more understandable this way.

Reply:
Good catch with the regex, I like your proposal.
Just to make it clear though, I think that you don't expect slashes anywhere in the `sed` expression, so although there is nothing wrong with pipe as delimiter, I see no acute reason for it. What do you think?
Next, you may simplify the quotes situation by unifying `"${BOOT_DEV}"' \2|'` to `"${BOOT_DEV} \2|"`.